src/SoundModule.py

```diff
@@ -0,0 +1,235 @@
+"""
+The SoundModule is a library of functions designed to perform real-time signal processing on blocks of audio.
+Each function takes inputs and outputs in a similar fashion, hence allowing them to
+be aggregated into a signal chain in an ad-hoc fashion.
+The various functions currently available in Sound-Module were developed keeping in mind
+their use in building the aesthetics necessary for SERIAL.
+https://github.com/reckoner165/TraceMelody
+
+Sumanth Srinivasan - December 2016
+"""
+
+import math
+import struct
+from math import cos
+from math import pi
+import random
+
+# Pans audio by applying different gains on left and right channels
+def pan_stereo(x, gain_L, gain_R):
+
+    x_stereo = [0 for n in range(0,2*len(x))]
+    if gain_L > 1 or gain_L < 0 or gain_R > 1 or gain_R < 0:
+        print "Invalid Gain. Try between 0 and 1"
```

Comment:
I have a doubt here - what will `pan_stereo(...)` do if the gains are invalid. Should it proceed with the flow with invalid gains? Or should it return immediately?

Reply:
ah, good catch. I was meaning to raise a `ValueError` there. Noted.

Reply:
If you're planning on raising a `ValueError`, how is it going to be handled downstream?

Reply:
If you're planning on raising a `ValueError`, it would be best if the list generation `x_stereo` is moved from line#21 to line#25.